subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultPluginRegistry.java

```diff
@@ -118,15 +118,19 @@ public PluginRegistry createChild(final ClassLoaderScope lookupScope) {
     @Nullable
     @Override
     public PluginImplementation<?> lookup(PluginId pluginId) {
-        PluginImplementation lookup;
+        PluginImplementation<?> lookup;
         if (parent != null) {
             lookup = parent.lookup(pluginId);
             if (lookup != null) {
                 return lookup;
             }
         }
 
-        return lookup(pluginId, classLoaderScope.getLocalClassLoader());
+        lookup = lookup(pluginId, classLoaderScope.getLocalClassLoader());
+        if (lookup != null) {
+            return lookup;
+        }
+        return lookup(pluginId, Thread.currentThread().getContextClassLoader());
```

Comment:
When script plugins are evaluated, their context classloader is bound to the current thread. One way to get this context visible to the plugin registry is to access that classloader if all else fails. Feels dirty, but works.

Reply:
I don't think we should do this. We should pass it in instead. We have actively removed assumptions like this in the past.